community-operators/spark-gcp/sparkoperator.2.4.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,292 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: sparkoperator.v2.4.0
+  namespace: placeholder
+  annotations:
```

Comment:
A few fields in annotations should be added to this operator:
`metadata.annotations.capabilities` should be added with the support-level for the operator. The 5 levels can be found in this diagram: https://www.operatorhub.io/operator/images/capability-level-diagram.svg
`metadata.annotations.description` should be added with a short one line description for the operator, with a limit of around 135 characters.
`metadata.annotations.containerImage` should be added with the operator's container image, which should match the image used in the deployment (ie: `gcr.io/spark-operator/spark-operator:v2.4.0-v1beta1-latest`)
`spec.maintainers` should be defined with a contact email as follows:
```
  maintainers:
  - name: Maintainer for spark-on-k8s
    email: maintainer@gcp.com
```
These fields are requested for displaying this operator properly in the OperatorHub UI.

Reply:
@SamiSousa Good catch, I've fixed the indentation for the annotation fields. And I've put down `capabilities: Basic Install` for now. I can upgrade the capability later if I can test out the upgrade capability of the operator.

Also for maintainers if you recall our discussion above we decided not to put down a maintainer since it's a Github open source project and not an official GCP product. So I'm not sure who's email to list.

Reply:
@hasbro17 I recall our previous conversation, but it's been decided that OperatorHub.io CSVs need to have certain fields to be rendered correctly in the UI. One of those fields is `maintainers` which needs to have a name and email that can be contacted for this operator.
However, there is a slack channel listed on the operator repo's Readme, which would be the ideal location for questions. I feel that including a link to this would be sufficient as a point of contact.
@dmesser @robszumski How should we handle this situation?

Reply:
@SamiSousa I've added the Slack channel link to the `spec.links` section. Not sure if I was supposed to add that to that annotation section.
Do you know if there's a list of accepted annotation tags so I know which ones to use?

Reply:
@SamiSousa I think if there are sufficient means for contact to the maintainers in the `links` section then it's fine to omit the maintainer field. Consequently, in the future we shall either relax that requirement and make `maintainers` info recommended but not required or support other data type there than email.